client/navigation/components/container/index.js

```diff
@@ -80,38 +80,42 @@ const Container = ( { menuItems } ) => {
 						) }
 					></NavigationBackButton>
 				) }
-				{ categories.map( ( category ) => (
-					<NavigationMenu
-						key={ category.id }
-						title={ category.title }
-						menu={ category.id }
-						parentMenu={ category.parent }
-						backButtonLabel={ category.backButtonLabel }
-					>
-						<NavigationGroup>
-							{ menuItems
-								.filter(
-									( item ) =>
-										item.parent === category.id &&
-										item.menuId !== 'secondary'
-								)
-								.map( ( item ) => (
-									<Item key={ item.id } item={ item } />
-								) ) }
-						</NavigationGroup>
-						<NavigationGroup>
-							{ menuItems
-								.filter(
-									( item ) =>
-										item.parent === category.id &&
-										item.menuId === 'secondary'
-								)
-								.map( ( item ) => (
-									<Item key={ item.id } item={ item } />
-								) ) }
-						</NavigationGroup>
-					</NavigationMenu>
-				) ) }
+				{ categories.map( ( category ) => {
+					const primaryItems = menuItems.filter(
+						( item ) =>
+							item.parent === category.id &&
+							item.menuId !== 'secondary'
+					);
+					const secondaryItems = menuItems.filter(
+						( item ) =>
+							item.parent === category.id &&
+							item.menuId === 'secondary'
+					);
```

Comment:
Is it worthwhile to use lodash's [partition](https://lodash.com/docs/4.17.15#partition) here to reduce looping through twice to one? Taking it a step further, should we memoize here as well? `menuItems` rarely changes, but we'll be looping on every render. Would this even have an effect in performance?

Reply:
I don't think we can use `partition` here since that will separate into 2 groups and ignore categories, but I came up with a reducer in https://github.com/woocommerce/woocommerce-admin/pull/5403/commits/3fa42b709023c80c60a040f37701b9dfd3ced93e that I think is pretty efficient.